submission/stats/src/main/kotlin/uk/ac/ebi/stats/service/SubmissionStatsService.kt

```diff
@@ -0,0 +1,43 @@
+package uk.ac.ebi.stats.service
+
+import uk.ac.ebi.stats.exception.StatAlreadyExistsException
+import uk.ac.ebi.stats.exception.StatNotFoundException
+import uk.ac.ebi.stats.mapping.SubmissionStatMapper
+import uk.ac.ebi.stats.model.SubmissionStat
+import uk.ac.ebi.stats.model.SubmissionStatType
+import uk.ac.ebi.stats.persistence.model.SubmissionStatDb
+import uk.ac.ebi.stats.persistence.repositories.SubmissionStatsRepository
+
+class SubmissionStatsService(private val submissionStatsRepository: SubmissionStatsRepository) {
+    fun findByType(submissionStatType: SubmissionStatType): List<SubmissionStat> =
+        submissionStatsRepository.findAllByType(submissionStatType).map { SubmissionStatMapper.toSubmissionStat(it) }
+
+    fun findByAccNoAndType(accNo: String, submissionStatType: SubmissionStatType): SubmissionStat =
+        SubmissionStatMapper.toSubmissionStat(
+            submissionStatsRepository.findByAccNoAndType(accNo, submissionStatType) ?:
+            throw StatNotFoundException(accNo, submissionStatType))
+
+    fun save(stat: SubmissionStat): SubmissionStat {
```

Comment:
I think save operation should both save or update which is the default behavior in jpa.

Reply:
Oh, I didn't know that one, cool.